pkg/frontend/querymiddleware/codec.go

```diff
@@ -180,6 +180,8 @@ func (prometheusCodec) MergeResponse(responses ...Response) (Response, error) {
 	}
 
 	promResponses := make([]*PrometheusResponse, 0, len(responses))
+	promWarningsMap := make(map[string]interface{}, 0)
```

Comment:
Using a `map[string]struct{}` here should be faster and use less memory: https://itnext.io/set-in-go-map-bool-and-map-struct-performance-comparison-5315b4b107b has some benchmarks and https://stackoverflow.com/a/22770744 has some other data that explains why.

Reply:
Thanks for the helpful links!